Gambot.Modules.Roll/RollResponseProducer.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Gambot.Core;
+using Gambot.Data;
+
+namespace Gambot.Modules.Roll
+{
+    public class RollResponseProducer : IMessageProducer
+    {
+        private IDataStore factoidDataStore;
+        private readonly IVariableHandler variableHandler;
+
+        private const string DefaultRollSuccessReply =
+            "<reply> $who rolled a $diceRoll.";
+        private const string DefaultRollFailedReply =
+            "<reply> That's a leaner.";
+
+        public RollResponseProducer(IVariableHandler variableHandler)
+        {
+            this.variableHandler = variableHandler;
+        }
+
+        public void Initialize(IDataStoreManager dataStoreManager)
+        {
+            factoidDataStore = dataStoreManager.Get("Factoids");
+        }
+
+        public ProducerResponse Process(IMessage message, bool addressed)
+        {
+            if (addressed)
+            {
+                var trimmedMessage = message.Text.Trim();
+                var match = Regex.Match(trimmedMessage, "roll (.+)");
+                if (match.Success)
+                {
+                    var diceSyntaxRegex = @"^(\d+)[dD](\d+)$|^(\d+)$";
+                    var rolls = match.Groups[1].Value.Split('+').Select(roll => roll.Trim());
```

Comment:
Looks like you only support joining expressings with "+," but there are use cases for "-" as well. For example, I recently started a campaign of Dungeon World. In that game, if your character has a noticeable wound, they take -1 forward to Charisma checks. That roll would be "2d6-1." If a Ranger gets a partial success on a Volley roll, their damage roll would be "1d8-1d6."